atomistics/workflows/evcurve/helper.py

```diff
@@ -0,0 +1,185 @@
+    return apply_strain(structure=structure, epsilon=strains, return_box=True)
+
+
+def apply_strain(
+    structure: Atoms, epsilon: float, return_box: bool = False, mode: str = "linear"
+) -> Atoms:
+    """
+    Apply a given strain on the structure. It applies the matrix `F` in the manner:
+
+    ```
+        new_cell = F @ current_cell
+    ```
+
+    Args:
+        epsilon (float/list/ndarray): epsilon matrix. If a single number is set, the same
+            strain is applied in each direction. If a 3-dim vector is set, it will be
+            multiplied by a unit matrix.
+        return_box (bool): whether to return a box. If set to True, only the returned box will
+            have the desired strain and the original box will stay unchanged.
+        mode (str): `linear` or `lagrangian`. If `linear`, `F` is equal to the epsilon - 1.
+            If `lagrangian`, epsilon is given by `(F^T * F - 1) / 2`. It raises an error if
+            the strain is not symmetric (if the shear components are given).
+    """
+    epsilon = np.array([epsilon]).flatten()
+    if len(epsilon) == 3 or len(epsilon) == 1:
+        epsilon = epsilon * np.eye(3)
+    epsilon = epsilon.reshape(3, 3)
+    if epsilon.min() < -1.0:
+        raise ValueError("Strain value too negative")
+    if return_box:
+        structure_copy = structure.copy()
+    else:
+        structure_copy = structure
+    cell = structure_copy.cell.copy()
+    if mode == "linear":
+        F = epsilon + np.eye(3)
+    elif mode == "lagrangian":
+        if not np.allclose(epsilon, epsilon.T):
+            raise ValueError("Strain must be symmetric if `mode = 'lagrangian'`")
+        E, V = np.linalg.eigh(2 * epsilon + np.eye(3))
+        F = np.einsum("ik,k,jk->ij", V, np.sqrt(E), V)
+    else:
+        raise ValueError("mode must be `linear` or `lagrangian`")
+    cell = np.matmul(F, cell)
+    structure_copy.set_cell(cell, scale_atoms=True)
+    if return_box:
+        return structure_copy
+
+
+def get_energy_lst(output_dict: dict, structure_dict: dict) -> list:
+    return [output_dict["energy"][k] for k in structure_dict.keys()]
+
+
+def get_volume_lst(structure_dict: dict) -> list:
+    return [structure.get_volume() for structure in structure_dict.values()]
+
+
+def fit_ev_curve_internal(
+    volume_lst: np.ndarray, energy_lst: np.ndarray, fit_type: str, fit_order: int
+) -> EnergyVolumeFit:
+    fit_module = EnergyVolumeFit(
+        volume_lst=volume_lst,
+        energy_lst=energy_lst,
+    )
+    fit_module.fit(fit_type=fit_type, fit_order=fit_order)
+    return fit_module
+
+
+def fit_ev_curve(
+    volume_lst: np.ndarray, energy_lst: np.ndarray, fit_type: str, fit_order: int
+) -> dict:
+    return fit_ev_curve_internal(
+        volume_lst=volume_lst,
+        energy_lst=energy_lst,
+        fit_type=fit_type,
+        fit_order=fit_order,
+    ).fit_dict
+
+
+def get_strains(
+    vol_range: Optional[float],
+    num_points: Optional[int],
```

Comment:
Doesn't look optional

Reply:
It is either `vol_range` and `num_points` or `strain_lst`.

Reply:
It categorically is not. The code as written does not allow `vol_range` and `num_points` to be optional. It is `vol_range` and `num_points`, or `vol_range` and `num_points` (both ignored) and `strain_list`. 

```python
>>> from atomistics.workflows.evcurve.helper import get_strains
>>> 
>>> get_strains(strain_lst=[1,2,3])
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[6], line 3
      1 from atomistics.workflows.evcurve.helper import get_strains
----> 3 get_strains(strain_lst=[1,2,3])

TypeError: get_strains() missing 2 required positional arguments: 'vol_range' and 'num_points'
```

You've placed two purely positional arguments with no default and then hinted them as optional.

Reply:
This is now addressed in https://github.com/pyiron/atomistics/pull/252